File: 이진영/프로그래머스/두 큐 합 같게 만들기.js
function solution(queue1, queue2) {
    const totalq = [...queue1, ...queue2];
    const target = totalq.reduce((p, n) => p + n, 0) / 2;
    
    if (!Number.isInteger(target)) return -1;
    
    let count = 0;
    let left = 0;
    let right = queue1.length;
    let sum = queue1.reduce((p, n) => p + n, 0);
    
    while (count < totalq.length * 2) {
        if (sum === target) return count;

        if (sum > target) {
            sum -= totalq[left++];
        } else {
            sum += totalq[right % totalq.length];
            right++;
        }
        count++;
    }
    
    return -1;
}

File: 정민선/프로그래머스/두 큐 합 같게 만들기.py
from collections import deque

def solution(queue1, queue2):
    target = (sum(queue1) + sum(queue2))//2
    
    array1, array2 = deque(queue1), deque(queue2)
    sum1, sum2 = sum(array1), sum(array2)
    
    loop = len(queue1) * 2 + len(queue2) * 2
    count = 0
    
    while count <= loop:
        if sum1 == target:
            return count
        
        if sum1 > sum2:
            x = array1.popleft()
            array2.append(x)
            sum1 -= x
            sum2 += x
        else:
            x = array2.popleft()
            array1.append(x)
            sum1 += x
            sum2 -= x
        count += 1
    
    return -1